source/overloading.tex

```diff
@@ -1529,21 +1530,84 @@
+whose \grammarterm{defining-type-id}
+is a \grammarterm{simple-template-id} \tcode{B<L>},
+the guides of \tcode{A} are the set of function or function templates
+formed as follows.
+For each function or function template \tcode{f} in the guides of \tcode{B},
+form a function or function template \tcode{f'}
+according to the following procedure and add it to the set:
+\begin{itemize}
+\item
+  Deduce the template arguments of the return type of \tcode{f}
+  from \tcode{B<L>} according to the process in\iref{temp.deduct.type}
+  with the exception that deduction does not fail
+  if not all template arguments are deduced.
+  Let \tcode{g} denote the result of substituting
+  these deductions into \tcode{f}.
+  If the substitution fails, no \tcode{f'} is produced.
+  Form the function or function template \tcode{f'} as follows:
+  \begin{itemize}
+  \item
+    The function type of \tcode{f'} is the function type of \tcode{g}.
+  \item
+    If \tcode{f} is a function template,
+    the template parameter list of \tcode{f'} consists of
+    all the template parameters of \tcode{A}
+    (including their default template arguments)
+    that appear in the above deductions or
+    (recursively) in their default template arguments,
+    followed by the template parameters of \tcode{f} that were not deduced
+    (including their default template arguments).
+  \item
+    The associated constraints\iref{temp.constr.decl} are
+    the conjunction of the associated constraints of \tcode{g} and
+    a constraint that is satisfied if and only if
+    the arguments of \tcode{A} are deducible (see below) from the return type.
+  \end{itemize}
+\item
+  If \tcode{f} is a copy deduction candidate\iref{over.match.class.deduct},
+  then \tcode{f'} is considered to be so as well.
+\item
+  If \tcode{f} was generated
+  from a deduction-guide\iref{over.match.class.deduct},
+  then \tcode{f'} is considered to be so as well.
+\item
+  The \grammarterm{explicit-specifier} of \tcode{f'} is
+  the \grammarterm{explicit-specifier} of g (if any).
+\end{itemize}
+
+\pnum
+The arguments of a template \tcode{A} are said to be
+deducible from a type \tcode{T} if, given a class template
+\begin{codeblock}
+template <typename> class AA;
+\end{codeblock}
+with a single partial specialization
+whose template parameter list is that of \tcode{A} and
+whose template argument list is a specialization of \tcode{A}
+with the template argument list of \tcode{A}\iref{temp.dep.type},
+\tcode{AA<T>} matches the partial specialization.
+
 \pnum
 Initialization and overload resolution are performed as described
 in \ref{dcl.init} and \ref{over.match.ctor}, \ref{over.match.copy},
 or \ref{over.match.list} (as appropriate for the type of initialization
 performed) for an object of a hypothetical class type, where
-the selected functions and function templates are considered to be the
+the guides of the template named by the placeholder are considered to be the
```

Comment:
What placeholder?

Reply:
If you use a "template-name" where a type is expected, that template-name is a placeholder per dcl.type.simple and dcl.type.class.deduct. Hm... We don't use placeholder-type-specifier for these.

Reply:
At least add a reference?  And maybe open an issue as well?

Reply:
p1 and p2 prominently talk about "placeholder", so... no from me.